tools-local/unix/probe-tool.sh

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env bash
+
+usage()
+{
+    echo "usage: $0 <repository-root> <tool-name> <override-scripts-folder-path> <strict-tool-version-match>"
+    echo "repository-root                   Path to repository root."
+    echo "tool-name                         Name of the tool to search and/or download."
+    echo "override-scripts-folder-path      If a path is specified then, search and acquire scripts from the specified folder will be invoked. Otherwise, search will use the default search and acquire scripts located within the repository."
+    echo "strict-tool-version-match         If equals to \"strict\" then, search will ensure that the version of the tool searched is the declared version. Otherwise, search will attempt to find a version of the tool, which may not be the declared version."
+    echo ""
+    echo "Invokes an extension that calls the appropriate search and/or acquire scripts. ToolName and StrictToolVersionMatch are passed on to the extension."
+    echo ""
+    echo "Example #1"
+    echo "probe-tool.sh \"/Users/dotnet/corefx\" cmake "" """
+    echo "Probes for CMake, not necessarily the declared version, using the default search and acquire scripts located within the repository."
+    echo ""
+    echo "Example #2"
+    echo "probe-tool.sh \"/Users/dotnet/corefx\" cmake "" strict"
+    echo "Probes for the declared version of CMake using the default search and acquire scripts located within the repository."
+    echo ""
+    echo "Example #3"
+    echo "probe-tool.sh \"/Users/dotnet/corefx\" cmake \"/Users/dotnet/MyCustomScripts\" strict"
+    echo "Probes for the declared version of CMake using the search and acquire scripts located in \"/Users/dotnet/MyCustomScripts\"."
+    echo ""
+}
+
+if [ $# -ne 4 ]; then
+    usage
+    exit 1
+fi
+
+repoRoot="$(cd "$1"; pwd -P)"
+toolName="$2"
+overrideScriptsFolderPath="$3"
+strictToolVersionMatch="$4"
+
+scriptPath="$(cd "$(dirname "$0")"; pwd -P)"
+. "$scriptPath/tool-helper.sh"
+
+exit_if_arg_empty "repository-root" "$repoRoot"
+exit_if_arg_empty "tool-name" "$toolName"
+
+if [ ! -z "$overrideScriptsFolderPath" ] && [ ! -d "$overrideScriptsFolderPath" ]; then
```

Comment:
As discussed offline, validate the given value is a valid directory only if a non-empty string (path) is provided. 

---
In reply to: [111428605](https://github.com/ravimeda/corefx/pull/9#discussion_r111428605) [](ancestors = 111428605)